cluster-state-service/handler/run/run.go

```diff
@@ -92,15 +92,27 @@ func StartClusterStateService(queueName string, bindAddr string, etcdEndpoints [
 	// start event processor
 	processor := event.NewProcessor(stores)
 
-	sqsClient := clients.NewSQSClient(awsSession)
+	if len(queueName) > 0 {
+		sqsClient := clients.NewSQSClient(awsSession)
 
-	// start event consumer
-	consumer, err := event.NewConsumer(sqsClient, processor, queueName)
-	if err != nil {
-		return errors.Wrapf(err, "Could not start the consumer")
-	}
+		// start event consumer
+		consumer, err := event.NewSQSConsumer(sqsClient, processor, queueName)
+		if err != nil {
+			return errors.Wrapf(err, "Could not start the consumer")
+		}
+
+		go consumer.PollForEvents(ctx)
+	} else {
```

Comment:
Should check for `len(streamName) > 0` for the kinesis path; else should return a "queue or stream required" error